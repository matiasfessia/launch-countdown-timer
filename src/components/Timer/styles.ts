import { css } from '@emotion/react';

export const styles = {
  cardsContainer: css`
    display: flex;
    justify-content: space-around;
  `
};
