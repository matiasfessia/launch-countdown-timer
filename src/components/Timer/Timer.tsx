/** @jsxImportSource @emotion/react */
import { useState, useEffect } from 'react';

import { styles } from './styles';
import { FlipCard } from '../FlipCard';

export const Timer = () => {
  const launchDate = new Date('Apr 8, 2021 21:30:30').getTime();
  const [timeleft, setTimeLeft] = useState<number>(launchDate - new Date().getTime());

  // @todo maybe is a good idea to create a custom hook for the following interval (dan's post about it).
  useEffect(() => {
    const interval = setInterval(() => {
      setTimeLeft(launchDate - new Date().getTime());
    }, 1000);
    return () => clearInterval(interval);
  }, [launchDate]);

  // @todo create helpers to parse the timedate to seconds, minutes, hours and days.
  return (
    <div css={styles.cardsContainer}>
      <FlipCard title="days" value={String(Math.floor(timeleft / (1000 * 60 * 60 * 24))).padStart(2, '0')} />
      <FlipCard title="hours" value={String(Math.floor((timeleft % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60))).padStart(2, '0')} />
      <FlipCard title="minutes" value={String(Math.floor((timeleft % (1000 * 60 * 60)) / (1000 * 60))).padStart(2, '0')} />
      <FlipCard title="seconds" value={String(Math.floor((timeleft % (1000 * 60)) / 1000)).padStart(2, '0')} />
    </div>
  );
};

export default Timer;
