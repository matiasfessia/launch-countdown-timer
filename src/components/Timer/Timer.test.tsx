import { render } from '@testing-library/react';
import { Timer } from './Timer';

it('renders correctly', () => {
  const { container } = render(<Timer />);
  expect(container).toMatchSnapshot();
});
