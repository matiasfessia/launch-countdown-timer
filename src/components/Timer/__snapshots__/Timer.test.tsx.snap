// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-1xpqvqu-Timer"
  >
    <div
      class="css-y5ro21-FlipCard"
      data-value="00"
    >
      <div
        class="css-nq5f8h-FlipCard"
        data-value="00"
      />
      <div
        class="css-xp1t8d-FlipCard"
      />
      <div
        class="css-18hlcob-FlipCard"
      />
    </div>
    <div
      class="css-y5ro21-FlipCard"
      data-value="21"
    >
      <div
        class="css-nq5f8h-FlipCard"
        data-value="21"
      />
      <div
        class="css-xp1t8d-FlipCard"
      />
      <div
        class="css-18hlcob-FlipCard"
      />
    </div>
    <div
      class="css-y5ro21-FlipCard"
      data-value="11"
    >
      <div
        class="css-nq5f8h-FlipCard"
        data-value="11"
      />
      <div
        class="css-xp1t8d-FlipCard"
      />
      <div
        class="css-18hlcob-FlipCard"
      />
    </div>
    <div
      class="css-y5ro21-FlipCard"
      data-value="25"
    >
      <div
        class="css-nq5f8h-FlipCard"
        data-value="25"
      />
      <div
        class="css-xp1t8d-FlipCard"
      />
      <div
        class="css-18hlcob-FlipCard"
      />
    </div>
  </div>
</div>
`;
